### PR TITLE
Dependencies: Bump github.com/grafana/grafana-plugin-sdk-go from v0.197.0 to v0.198.0

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Bumps plugin-related dependencies:

- github.com/grafana/grafana-plugin-sdk-go from v0.197.0 to v0.198.0

[As a side effect](https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.198.0), those dependencies are also bumped:
- google.golang.org/protobuf from 1.31.0 to 1.32.0
- google.golang.org/grpc from 1.59.0 to 1.60.1
- github.com/hashicorp/go-hclog from v1.6.1 to v1.6.2
- github.com/prometheus/client_golang from v1.17.0 to v1.18.0
- github.com/google/uuid from v1.4.0 to v1.5.0

Because this bumps the google.golang.org/protobuf module, I have also re-generated the protobuf files via `make protobuf`.

`make protobuf` has also been updated to work with the latest `protoc` and `protoc-gen-go`.

There are also some changes to `pkg/plugins/backendplugin/pluginextensionv2/generate.sh`. Without those changes, I get the following error:

```bash
make protobuf  
bash scripts/protobuf-check.sh  
bash pkg/plugins/backendplugin/pluginextensionv2/generate.sh  
--go_out: protoc-gen-go: plugins are not supported; use 'protoc --go-grpc_out=...' to generate gRPC  
  
See https://grpc.io/docs/languages/go/quickstart/#regenerate-grpc-code for more information.  
make: *** [Makefile:308: protobuf] Error 1
```

Tested with:

- [x] Plugin built with old SDK and Grafana with upgraded dependencies
- [x] Plugin built with new SDK and Grafana with upgraded dependencies
- [x] Plugin built with new SDK and Grafana without upgraded dependencies

**Why do we need this feature?**

Dependencies updates.

**Who is this feature for?**

Everyone.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/backendplugin/pluginextensionv2/generate.sh [hunk 1]
```diff
@@ -13,4 +13,4 @@ DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
 cd "$DIR"
 
-protoc -I ./ *.proto --go_out=plugins=grpc:./
+protoc -I ./ *.proto --go_out=. --go-grpc_out=.
```
Comment (xnyo) [inline pkg/plugins/backendplugin/pluginextensionv2/generate.sh:16]: I couldn't re-generate protobuf files without this change.

This is also the same command we use to generate the protobuf files in the SDK:

https://github.com/grafana/grafana-plugin-sdk-go/blob/6ecd6938cfed8cbfe1cdd1a43696321e23c362d1/proto/generate.sh#L16-L19

**Review by wbrowne — COMMENTED** [review]
Just wanted to confirm that the relevant testing has been done on the SDK side considering the more "risky" package updates like gRPC, protobuf, etc.

And then for this PR, did you test the renderer and secrets manager? Happy to do so, but just wanted to check first 😄

**Comment by xnyo** [comment]
> Just wanted to confirm that the relevant testing has been done on the SDK side considering the more "risky" package updates like gRPC, protobuf, etc.
> 
> And then for this PR, did you test the renderer and secrets manager? Happy to do so, but just wanted to check first 😄

I have tested it with plugins built with the old SDK as well as the new SDK, and the new SDK against old Grafana as well and all those scenarios are working as expected.

I have tested the renderer by building Enterprise and rendering a panel to PNG and it is works fine.

I have enabled the secrets manager in config ini but still not sure how to test it properly.

**Review by wbrowne — APPROVED** [review]
That's good enough for me! 🎉

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**